src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -1246,6 +1246,17 @@ public Tensor pin_memory()
                 return new Tensor(res);
             }
 
+            /// <summary>
+            /// This function will set the `tensor.grad()` attribute to a custom tensor. Make sure to detach the tensor from
```

Comment:
Should the detach be taken care of here, then, perhaps? Of should it be transferred to the dispose scope of the tensor the grad belongs to?

Reply:
Sure, will transfer to the dispose scope of the tensor.

Reply:
I don't see an easy way to do that. How would I do that?
For now, just applying the detach.

Reply:
How about adding a method to DisposeScope for a direct transfer rather than transfer-to-parent?

Reply:
Sure, but perhaps we should keep this in a different release?
I'd rather not mess with the dispose scopes right before you're off.

Reply:
If you think it's a good idea, I put it in code. It's currently in a PR on my fork into this branch. https://github.com/shaltielshmid/TorchSharp/pull/2

If this draft looks good to you, I'll merge it in and we can put it into this version.